src/vcsparser.core/Measure.cs

```diff
@@ -2,15 +2,19 @@
 
 namespace vcsparser.core
 {
-    public class Measure
+
+    public abstract class Measure

```

Comment:
Got it. I forgot about SonarMeasureJson and friends.

Would you mind going for IMeasure then? That would require changing the consumers, but it is doable (I've tested it locally). The reason is that I'd like to avoid the abstract class since it can encourage inheritance depth and behaviour being added at Measure level. 

```
    public interface IMeasure
    {
        string MetricKey { get; }

        string File { get; }
    }

    public class Measure<T> : IMeasure
    {
        [JsonProperty("metric-key")]
        public string MetricKey { get; set; }

        [JsonProperty("file", NullValueHandling = NullValueHandling.Ignore)]
        public string File { get; set; }

        [JsonProperty("value")]
        public T Value { get; set; }
    }
```